src/dmd/expressionsem.d

```diff
@@ -2052,8 +2052,14 @@ private bool functionParameters(const ref Loc loc, Scope* sc,
             // However, existing `@nogc` code may rely on it, so still do it when the parameter is explicitly marked `scope`
             const explicitScope = p.isLazy() ||
                 ((p.storageClass & STC.scope_) && !(p.storageClass & STC.scopeinferred));
+
+            // https://issues.dlang.org/show_bug.cgi?id=23175
+            // Don't trust `scope` from `in` while `-preview=in` transition is in progress
+            const scopeFromIn = global.params.previewIn && !(p.storageClass & STC.scope_) && (p.storageClass & STC.in_);
+
             if ((pStc & (STC.scope_ | STC.lazy_)) &&
                 ((global.params.useDIP1000 == FeatureState.enabled) || explicitScope) &&
+                !scopeFromIn &&
```

Comment:
Isn't the actual problem here that the DIP1000 check is too liberal and fails to detect this scope violation without `@safe`?